### ISSUE TITLE
Time profiling result is inconsistent with result from original caffe SSD

### ISSUE THREAD

**Issue by andyhahaha** [amdegroot/ssd.pytorch#24]
I wondering the time consumption at each part. (VGG, Extra, multi_box, detection)
From the result of caffe version, the VGG part accounts for up to 80 percent of time consumption.
However, in this version, the distribution of time consumption is as follow:

Total time : 0.018(seconds) per image
VGG part 8.4%
Extra layer 2.8%
Multi_box 61%
detect  27.5%
Most of time is from Multi_box and detect.
I measure it by python time.time() 

And both total time for one image is almost the same.
caffe : 19ms
pytorch : 18ms

I wondering why this inconsistence happen?

**Comment by amdegroot** [comment]
To be honest I'm not really sure. That's really cool that the time profile is actually better with pytorch though, I did not realize that.  It must have something to do with the way pytorch is designed under the hood with respect to gpu acceleration (I haven't been able to keep up with pytorch development as much recently :( ).

**Comment by markusnagel** [comment]
@andyhahaha I'm curious, how do you get the 18ms? Is that on one specific image or the average over many images (VOC test set)?

I get on average over VOC test set only 70-80 ms per image (with a GTX 1080). Though the variance is very high, some images need only 20ms, others over 100ms. It also seems that the same images take always (approximately) the same time.

I'm also wondering how @amdegroot gets the 45 FPS, is that the average or for one test image? It seems I should get at least the same or faster with my GTX 1080. Could you maybe also share some numbers on training time? Mine is currently at +-130ms per iteration (again on Pascal VOC with the provided training code).
